substrate/client/rpc/src/state/state_full.rs

```diff
@@ -450,11 +457,7 @@ where
 			.chain(storage_stream)
 			.filter(|storage| future::ready(!storage.changes.is_empty()));
 
-		let fut = async move {
-			sink.pipe_from_stream(stream).await;
-		};
-
-		self.executor.spawn("substrate-rpc-subscription", Some("rpc"), fut.boxed());
+		spawn_subscription_task(&self.executor, pipe_from_stream(pending, stream, 16));
```

Comment:
This is more clear to me, since its also explicit that we accept the subscription (something that was done under the hood by the jsonrpsee before).

I would probably add `rpc/lib/utils const DEFAULT_PENDING_BUFFER =16`, since that might change after more data is available to us.
Another approach here would be removing the capacity from `spawn_subscription_task`, then adding another `spawn_subscription_task_with_capacity` for more explicit use-cases. Leaving this up to you, looks good as it already

Reply:
Deal but let's wait for other review comments and whether it makes sense to increase or decrease this for certain subscriptions.

Reply:
Did we gain any insight from burnins about whether any connections were being swiftly closed because of this buffer size?

Esp because we buffer while waiting for the client to accept, it feels not unlikely that some subscriptions might immediately hit this (thinking eg chainHead_follow, or possibly storage subscriptions or something)

Reply:
But I agree that we could have a constant for the "default size" and then in a few cases we may use a bigger value.

Reply:
> Did we gain any insight from burnins about whether any connections were being swiftly closed because of this buffer size?

Some where closed but no real data or anything to draw any conclusions on it but looked good to me but
it's been a while so I might have forgotten.

> Esp because we buffer while waiting for the client to accept, it feels not unlikely that some subscriptions might immediately hit this (thinking eg chainHead_follow, or possibly storage subscriptions or something)

We had so little traffic on the new rpc methods and pretty much the behavior only concern the old rpc subscriptions (storage subscription in particular)